File: backend/app/schemas/tutor.py
from typing import Literal

from pydantic import BaseModel, Field


class TutorMessage(BaseModel):
    role: Literal["user", "assistant"]
    content: str


class TutorRequest(BaseModel):
    question: str = Field(min_length=1, max_length=2000)
    topic_slug: str | None = None
    # Prior turns in this conversation, oldest first. Not persisted anywhere
    # server-side — the frontend owns the running transcript and resends it
    # each time, which is what "multi-turn, no backend session state" means.
    history: list[TutorMessage] = Field(default_factory=list)


class TutorResponse(BaseModel):
    answer: str


class TutorStatus(BaseModel):
    enabled: bool

File: frontend/src/types/tutor.ts
export interface TutorMessage {
  role: "user" | "assistant";
  content: string;
}

export interface TutorRequest {
  question: string;
  topic_slug?: string | null;
  history: TutorMessage[];
}

export interface TutorResponse {
  answer: string;
}

export interface TutorStatus {
  enabled: boolean;
}
